cibuildwheel/platform_backend.py

```diff
@@ -0,0 +1,213 @@
+import os
+import shutil
+import sys
+from abc import abstractmethod
+from contextlib import contextmanager
+from pathlib import Path, PurePath
+from typing import Dict, Iterator, List, Optional
+
+from cibuildwheel.bashlex_eval import local_environment_executor
+from cibuildwheel.typing import Final, Literal, PathOrStr, PlatformName
+from cibuildwheel.util import call, ensure_virtualenv, shell
+
+
+class PlatformBackend:
```

Comment:
Shouldn't this be an ABC? (either through the ABCMeta metaclass or inheriting from abc.ABC)?

Reply:
done.